src/main/scala/com/databricks/spark/sql/perf/mllib/data/dataGeneration.scala

```diff
@@ -189,3 +214,59 @@ class GaussianMixtureDataGenerator(
   override def copy(): GaussianMixtureDataGenerator =
     new GaussianMixtureDataGenerator(numCenters, numFeatures, seed)
 }
+
+
+class RandStringGenerator(
+    val minLen: Int,
+    val maxLen: Int,
+    val charSet: String) extends RandomDataGenerator[String] {
+
+  private val rng = new java.util.Random()
+
+  override def nextValue(): String = {
+    val sb = new StringBuilder
+    var i = 0
+    val strLen = if (minLen == maxLen) minLen else {
+      minLen + rng.nextInt(maxLen - minLen + 1)
+    }
+    while (i < strLen) {
+      val nextChar = charSet.charAt(rng.nextInt(charSet.length))
+      sb.append(nextChar)
+      i += 1
+    }
+    sb.toString()
+  }
+
+  override def setSeed(seed: Long) {
+    rng.setSeed(seed)
+  }
+
+  override def copy(): RandStringGenerator = new RandStringGenerator(minLen, maxLen, charSet)
+}
+
+class OverlappedStringGenerator(
+    distinctCount: Int) extends RandomDataGenerator[String] {
+
+  private val rng = new java.util.Random()
+  private val distinctSetRng = new java.util.Random(123L)
+
+  @transient private var distinctSet: Array[String] = null
```

Comment:
Do we need to generate an in-memory set of distinct string values? This might not scale if `distinctCount` is large - maybe we could just generate random ints in (0, distinctCount] instead?